app/pages/4_Clean_Tables.py

```diff
@@ -99,40 +132,79 @@ def style_symbol(v, props=""):
     col7, col8, col9 = st.columns([1, 1, 1])
     with col7:
         total = st.checkbox(
-            "Calculate the Total of each columns, excluding the last row"
+            "Calculate the Total of each columns, excluding the last row", value=True
         )
+        country = st.checkbox("Activate the country filter", value=True)
+
     with col8:
         negativ = st.checkbox(
             "Show the negative numbers, for each columns detected as a numerical type"
         )
     with col9:
         symbol = st.checkbox(
-            "Show the cells that contain a special symbol : " + special_characters
+            "Show the cells that contain a special symbol : " + special_characters,
+            value=True,
         )
+        remove_symbols = st.checkbox("Remove the special symbols")
+
+    dataframe = edited_df.copy()
+
+    if remove_symbols:
+        pattern = "[" + re.escape(special_characters) + "]|\(.*?\)"
```

Comment:
Since the parentheses are in both conditions of the pattern and regex processes them from left to right, it will not check for the second condition when a parenthesis is matched in the first one.
A solution would be to change the conditions order: `"\(.*?\)|" + "[" + re.escape(special_characters) + "]"`
Examples
```
my_string = "42 (123)"
# With new pattern
re.findall("\(.*?\)|" + "[" + re.escape(special_characters) + "]", my_string)
>>> ['(123)']
# With old pattern
re.findall("[" + re.escape(special_characters) + "]" + "|\(.*?\)", my_string)
>>>['(', ')']
```

Reply:
thanks !